src/jit/importer.cpp

```diff
@@ -3534,55 +3578,95 @@ GenTreePtr Compiler::impIntrinsic(GenTreePtr            newobjThis,
 #ifndef LEGACY_BACKEND
         case CORINFO_INTRINSIC_Object_GetType:
         {
-            op1 = impPopStack().val;
+            JITDUMP("\n impIntrinsic: call to Object.GetType\n");
+            op1 = impStackTop(0).val;
 
             // If we're calling GetType on a boxed value, just get the type directly.
-            if (!opts.MinOpts() && !opts.compDbgCode)
+            if (op1->IsBoxedValue())
```

Comment:
Is this a change that always  impacts MinOpts codegen?

Is there a way to turn off this optimization using a ConfigDWORD?

Reply:
In the original code this check was redundant, so I removed it in the new code. No new behavior, we still won't recognize Type.GetType as an intrinsic in minopts/debug.

The check for minopts/debug is up at line 3386 in the new code.